src/Uecode/Bundle/ApiKeyBundle/Security/Authentication/Provider/ApiKeyProvider.php

```diff
@@ -58,12 +66,8 @@ public function authenticate(TokenInterface $token)
      * @return bool|ApiKeyUserToken
      * @throws AuthenticationException
      */
-    protected function doAuth(UserProviderInterface $provider, TokenInterface $token)
+    protected function doAuth($provider, TokenInterface $token)
     {
-        if (!method_exists($provider, 'loadUserByApiKey')) {
```

Comment:
Can you test this with multiple user provider? IIRC the reason i soft failed here (return false) was to skip this Provider, to go on to another provider